frontend/src/js/ui-components/InputDate.tsx

```diff
@@ -0,0 +1,114 @@
+import styled from "@emotion/styled";
+import { createElement, createRef, forwardRef, useState } from "react";
+import ReactDatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+import { formatDate, parseDate } from "../common/helpers/dateHelper";
+
+import BaseInput, { Props as BaseInputProps } from "./BaseInput";
+
+const Root = styled("div")`
+  position: relative;
+
+  .react-datepicker-wrapper {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    z-index: -1;
+  }
+  .react-datepicker-popper[data-placement^="bottom"] {
+    padding-top: 0.25rem;
+    transform: translate3d(0, 2rem, 0) !important;
+  }
+  .react-datepicker-popper[data-placement^="top"] {
+    padding-bottom: 0;
+    transform: translate3d(0, -2rem, 0) !important;
+  }
+`;
+
+const HiddenInput = styled("input")`
+  display: none;
+`;
+
+const StyledCalendar = styled("div")`
+  .react-datepicker__day--selected {
+    background: ${({ theme }) => theme.col.green};
+  }
+`;
+
+type Props = Omit<BaseInputProps, "inputType"> & {
+  value: string | null;
+  dateFormat: string;
+  className?: string;
+  onChange: (val: string | null) => void;
+  onCalendarSelect?: (val: string | null) => void;
+};
+
+const InputDate = forwardRef<HTMLInputElement, Props>(
+  (
+    {
+      className,
+      value,
+      dateFormat,
+      onChange,
+      onCalendarSelect,
+      onFocus,
+      ...props
+    },
+    ref,
+  ) => {
+    const datePickerRef = createRef<ReactDatePicker>();
+    const [focusBlocked, setFocusBlocked] = useState(false);
+
+    return (
+      <Root
+        className={className}
+        onKeyDown={(e) => {
+          if (e.key === "Escape") datePickerRef.current?.setOpen(false);
+        }}
+      >
+        <BaseInput
+          {...props}
+          ref={ref}
+          inputType="text"
+          value={value}
+          onChange={(val) => {
+            onChange(val as string);
+          }}
+          onFocus={(e) => {
+            if (focusBlocked) {
+              e.currentTarget.blur();
+              setFocusBlocked(false);
+            } else {
+              onFocus?.(e);
+              datePickerRef.current?.setOpen(true);
+            }
+          }}
+          inputProps={{
+            ...props?.inputProps,
+            onKeyPress: (e) => {
+              datePickerRef.current?.setOpen(false);
+              props.inputProps?.onKeyPress?.(e);
+            },
+          }}
+        />
+        <ReactDatePicker
+          ref={datePickerRef}
+          selected={value ? parseDate(value, dateFormat) : new Date()}
+          onChange={(val) => {
+            const selectedDate = formatDate(val as Date, dateFormat);
+            val && onChange(selectedDate);
```

Comment:
This confused me a bit: can `val` ever be null / undefined? If yes, `formatDate` might crash and we should replace the type cast with a null check. If no, we won't need the `val &&` check. Or am I missing something else here?

Reply:
Yes the onChange method passes a "Date | null" parameter.
I don't know in what case null is passed but to be sure I implemented a null check.